src/ipyradiant/visualization/explore/graph_explorer.py

```diff
@@ -0,0 +1,190 @@
+        super().__init__(*args, **kwargs)
+        label = label or "Available types:"
+        self.label.value = f"<b>{label}</b>"
+        self.children = tuple([self.label, self.select_widget])
+
+    def make_available_types(self):
+        if self.graph is not None:
+            ns = dict(self.graph.namespaces())
+            types = AllTypes.run_query(self.graph)
+            # TODO how to allow user specification of custom CustomURI class
+            self.select_widget.pithy_uris = tuple(
+                map(lambda x: CustomURI(x, namespaces=ns), types["o"].to_list())
+            )
+
+    @T.default("label")
+    def make_default_label(self):
+        return W.HTML("<b>Available types:</b>")
+
+    @T.default("select_widget")
+    def make_default_select_widget(self):
+        return SelectMultipleURI(rows=10)
+
+    @T.observe("graph")
+    def update_graph(self, change):
+        self.make_available_types()
+
+
+class RDFSubjectSelectMultiple(W.VBox):
+    # Define the query class
+    class SubjectsOfType(SPARQLQueryFramer, metaclass=MetaSubjectsOfType):
+        values = None  # note: query will not run without values
+
+    graph = T.Instance(Graph, allow_none=True).tag(default=None)
+    query = SubjectsOfType
+    _values = T.Instance(dict, allow_none=True).tag(default=None)
+    #
+    label = T.Instance(W.HTML)
+    select_widget = T.Instance(SelectMultipleURI)
+
+    def __init__(self, label: str = None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        label = label or "Available subjects:"
+        self.label.value = f"<b>{label}</b>"
+        self.children = tuple([self.label, self.select_widget])
+
+    @T.default("label")
+    def make_default_label(self):
+        # TODO update_label(label_str: str):
+        #  label should be a property that is based on label_text
+        return W.HTML("<b>Available subjects:</b>")
+
+    @T.default("select_widget")
+    def make_default_select_widget(self):
+        return SelectMultipleURI(rows=10)
+
+    def update_select(self):
+        if self.graph is not None and self._values is not None:
+            qres = self.run_query()
+            sw_uris = []
+            for uri, label in qres.values:
+                sw_uris.append(
+                    CustomItem(
+                        _repr=lambda x: f"{x.label}:   ->   {x.uri}",
+                        uri=uri,
+                        label=label,
+                    )
+                )
+            self.select_widget.pithy_uris = tuple(sw_uris)
+
+    @T.observe("graph")
+    def update_graph(self, change):
+        self.update_select()
+
+    @T.observe("_values")
+    def update_values(self, change):
+        if change.old != change.new:
+            self.query.values = change.new
+            self.update_select()
+
+    def run_query(self):
+        assert self.query.values is not None
+        return self.query.run_query(self.graph)
+
+
+class GraphExploreNodeSelection(W.VBox):
```

Comment:
Can you add a brief docstring?  And don't forget to have a space between the code and docstring (PEP8).